fedimint-bitcoind/src/lib.rs

```diff
@@ -211,8 +214,11 @@ impl DynBitcoindRpc {
                         .await;
 
                     match res {
-                        Ok(c) => {
-                            on_update(c);
+                        Ok(block_count) => {
+                            if last_block_count.load(Ordering::SeqCst) != block_count {
+                                on_update(block_count);
+                                last_block_count.store(block_count, Ordering::SeqCst);
+                            }
```

Comment:
Pretty sure this is only not racy because technically we don't even need atomics here. Why not just inline the `update_block_count` closure and use a normal mutable `u64` in the loop? Or even just give an argument to the closure, which is "last value" if the closure has some benefits I haven't considered.

Reply:
> Why not just inline the update_block_count closure and use a normal mutable u64 in the loop?

Some Send and Sync async nonsense.